lib/ansible/modules/database/mongodb/mongodb_replicaset.py

```diff
@@ -0,0 +1,407 @@
+#!/usr/bin/python
+
+# Copyright: (c) 2018, Rhys Campbell <rhys.james.campbell@googlemail.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = r'''
+---
+module: mongodb_replicaset
+short_description: Initialises a MongoDB replicaset before authentication has been turned on and then validates the configuration when it has been turned on.
+description:
+- Initialises a MongoDB replicaset before authentication has been turned on and then validates the configuration when it has been turned on.
+- Validation confirms the replicaset set name only.
+author: Rhys Campbell (@rhysmeister)
+version_added: "2.8"
+options:
+  login_user:
+    description:
+    - The username to authenticate with.
+    type: str
+  login_password:
+    description:
+    - The password to authenticate with.
+    type: str
+  login_database:
+    description:
+    - The database where login credentials are stored.
+    type: str
+    default: admin
+  login_host:
+    description:
+    - The MongoDB hostname.
+    type: str
+    default: localhost
+  login_port:
+    description:
+    - The MongoDB port to login to.
+    type: int
+    default: 27017
+  replica_set:
+    description:
+    - Replicaset name.
+    type: str
+    default: rs0
+  members:
+    description:
+    - A comma-separated string or a yaml list consisting of the replicaset members.
+    - Supply as a simple csv string, i.e. mongodb1:27017,mongodb2:27017,mongodb3:27017.
+    - If a port number is not provided then 27017 is assumed.
+    type: list
+  validate:
+    description:
+    - Performs some basic validation on the provided replicaset config.
+    type: bool
+    default: yes
+  ssl:
+    description:
+    - Whether to use an SSL connection when connecting to the database
+    type: bool
+    default: no
+  ssl_cert_reqs:
+    description:
+    - Specifies whether a certificate is required from the other side of the connection, and whether it will be validated if provided.
+    type: str
+    default: CERT_REQUIRED
+    choices: [ CERT_NONE, CERT_OPTIONAL, CERT_REQUIRED ]
+  arbiter_at_index:
+    description:
+    - Identifies the position of the member in the array that is an arbiter.
+    type: str
+  chainingAllowed:
```

Comment:
We tend to use snail_case, so all parameters should be snail_case.

Reply:
All changed. I was following MongoDB format but this way makes more sense.